erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -1666,6 +1668,7 @@ func (hph *HexPatriciaHashed) fold() (err error) {
 			fmt.Printf("formed leaf (%d %x, depth=%d) [%x] %s\n", row, nibble, depth, updateKey, cell.FullString())
 		}
 	case updateKindBranch:
+		CurrentCommitmentMetrics.UpdateBranch++
```

Comment:
not good place to do so. Check `CollectUpdate` - update may be skipped if equals to previous

Reply:
inside `collectUpdate`